bobtask/artifact_extract.go

```diff
@@ -135,8 +142,9 @@ func (t *Task) ArtifactExtract(artifactName hash.In, invalidFiles map[string][]t
 // shouldFetchFromCache checks if a file should be brought back from cache inside the target
 // A file will be brought back from cache if it's missing or was changed
 func shouldFetchFromCache(filename string, invalidFiles map[string][]target.Reason) bool {
+	// FIXME: accessing a hashmap twice is inefficent.
 	if _, ok := invalidFiles[filename]; !ok {
-		return false
+		return true
```

Comment:
@rdnt If files from the artifact are not present in  invalidFiles  we have to extract them.

Reply:
🙈